### PR TITLE
Fix Modifier deck for Safari

### PR DESCRIPTION
**Pull request by GinoGalotti**
I think I have it! On safari 10.1 I can see all the buttons, they work, and I can see the clip-path from drawing two cards. It should be supported from iOS Safari 9.3... but I don't have an iPhone to test.

Please, give it a go and tell me! Lets fix this.

---
This change is [Reviewable](https://reviewable.io/reviews/johreh/gloomycompanion/41)

### CODE REVIEW

File: cards.css [hunk 1]
```diff
@@ -174,6 +174,8 @@
     background-position:        100% 50%;
     background-size:            contain;
     background-image:           url(images/draw-two.svg);
+    cursor:                     pointer;
+
```
Comment (mwinckler) [inline cards.css]: Extra newline

File: cards.css [hunk 2]
```diff
@@ -338,13 +340,13 @@
 .card.modifier.front.left img.cover
 {
     left:                       -25%;
-    clip-path:                  inset(0% 24.5% 0% 24.5%);
+    -webkit-clip-path:                  inset(0% 24.5% 0% 24.5%);
```
Comment (mwinckler) [inline cards.css]: Indentation here and below, and I think we should keep the `clip-path` declaration in addition to (after) the `-webkit-clip-path` for non-prefixed support.

File: logic.js [hunk 3]
```diff
@@ -233,11 +233,11 @@ function refresh_ui()
             tableau.style.fontSize  = font_pixel_size + "px";
             break;
         }
-        
+
```
Comment (mwinckler) [inline logic.js]: Extra newline

File: logic.js [hunk 4]
```diff
@@ -303,7 +303,7 @@ function draw_ability_card(deck)
 {
     if (deck.must_reshuffle())
     {
-        reshuffle(deck);
+        reshuffle(deck, include_discards=true);
```
Comment (mwinckler) [inline logic.js]: This should just be `true`

File: logic.js [hunk 5]
```diff
@@ -328,7 +328,7 @@ function prevent_pull_animation(deck)
 function reshuffle_modifier_deck(deck)
 {
     deck.clean_discard_pile();
-    reshuffle(deck);
+    reshuffle(deck, include_discards=true);
```
Comment (mwinckler) [inline logic.js]: Same - just `true`

File: logic.js [hunk 6]
```diff
@@ -580,7 +580,7 @@ function apply_deck_selection(decks, preserve_existing_deck_state)
         container.appendChild(deck_space);
 
         place_deck(deck, deck_space);
-        reshuffle(deck);
+        reshuffle(deck, include_discards=true);
```
Comment (mwinckler) [inline logic.js]: Same - just `true`

File: logic.js [hunk 7]
```diff
@@ -685,7 +685,7 @@ function add_modifier_deck(container, deck)
     container.appendChild(modifier_container);
 
     place_deck(deck, deck_space);
-    reshuffle(deck);
+    reshuffle(deck, include_discards=true);
```
Comment (mwinckler) [inline logic.js]: Same - just `true`

File: cards.css (hunk 2 above)
Comment (GinoGalotti) [inline cards.css]: Are they useful? I tried on Chrome + Safari and both worked just with -webkit-clip-path... but I don't mind having it duplicated!

**Comment by GinoGalotti** [comment]
Changes made. Will merge later! Didn't know about keeping clip-path, but apparently, Visual Studio code was also recommending me to do so

**Comment by mwinckler** [comment]
In theory, the reason vendor prefixes like `-webkit-` exist is so that browser vendors can start implementing new/experimental features while the W3C spec is still in flux (or before any spec exists). At that stage browser implementations and syntax may vary, hence the prefixes. Then once the spec is done browsers implement the actual property, to spec, with no prefix.

In reality I don't think it's always that clean, but the idea is with the "spec" property name last in the CSS, it should override and vendor prefixes for browser versions that support the spec, and older browsers released with only prefix support continue to work.

**Pull request merged**